### ISSUE TITLE
7 bytes of the VP8 payload unencrypted for non key frames

### ISSUE THREAD

**Issue by lgrahl** [signalapp/ringrtc#38]
There's an [open todo you probably don't want left unresolved in production](https://github.com/signalapp/ringrtc/blob/46fe55060f18f958352a1a3e2f9bed766986ab98/src/rust/src/core/group_call.rs#L2422:L2431).

AFAIK this means that the first 7 bytes of the VP8 payload after the header are usually left unencrypted. The fix should be very simple by looking at the P bit (as [done by your SFU](https://github.com/signalapp/Signal-Calling-Service/blob/3b91b08a5ce2be25da8e13e06e6928474adf58b2/src/vp8.rs#L114)) to determine whether to leave 3 or 10 bytes unencrypted.

**Comment by ZagButNoZig** [comment]
@jrose-signal any updates on this? It still seems to be on main one year later. Is this a security/privacy concern?

**Comment by awaitlink** [comment]
Looks like there's some movement to make it encrypted: https://github.com/signalapp/webrtc/pull/164.

**Comment by lgrahl** [comment]
> Looks like there's some movement to make it encrypted: [signalapp/webrtc#164](https://github.com/signalapp/webrtc/pull/164).

Nah, that's just audio atm (although I'm not sure why one would need to leave parts of an audio frame unencrypted).

**Comment by hackerbirds** [comment]
I believe https://github.com/signalapp/ringrtc/commit/89ccd855521c834b365976561c14a77e7d6cd66a fixed this problem?

**Comment by jim-signal** [comment]
> I believe [89ccd85](https://github.com/signalapp/ringrtc/commit/89ccd855521c834b365976561c14a77e7d6cd66a) fixed this problem?

Yes, thanks @hackerbirds, that does indeed complete the sequence of commits that removed this. Closing.

**Comment by lgrahl** [comment]
Out of curiosity, how do you signal the spatial resolution of a video stream now? Is it an additional header extension?

**Comment by jim-signal** [comment]
Yes, we use the [dependency descriptor](https://aomediacodec.github.io/av1-rtp-spec/#a8-dependency-descriptor-format) header extension now.
